### ISSUE TITLE
I need to implement GPSToggler in my project.

### ISSUE THREAD

**Issue by suyashcyber94** [sms2000/GPSToggler#3]
I have a project for tracking device in which I have to implement GPSToggler to turn on GPS remotely and I am new in android development so I am not understanding this code properly.

my project works like I will send SMS to the device and it will turn on mobile DATA and GPS after receiving that SMS and then I can track the device remotely. I am able to turn on DATA but not the GPS. So, It will be really helpful for me if you will guide me how to implement this thing into my project. 
I can also share my project if you want something regarding this.

Thanks in advance.

**Comment by sms2000** [comment]
Hi

1st you should understand and accept the fact you cannot control the GPS on/off state from the user level application.
Only the system app can do it.

You can turn your app to a system app but you need rooted devices
.
If you agree you may implement and I'm ready to assist.

**Comment by suyashcyber94** [comment]
Yes I am doing this on rooted device and Yes I will turn my app to system app.
But I will need your help in converting to system app too because of my lack of knowledge.

Thanks

**Comment by sms2000** [comment]
Particularly in this project I implemented a partial "systimizer": the main application is a user-mode one and a small intnet-driven stub application systemized. 

To systemyze any app (including the self) you need to to do the following:

1) Obtain root (see the MainActivity.java file, "flushCommnad" function).
2) Locate you APK file in the file system.
3) Make the /system filesystem writable (mount -rw ... command)
4) Move this APK to the /system/priv-apk or /system/app if the 1st one doesn't exist.
5) Reboot or ask the user to reboot. 

All the low-level commands like "mount", "cp" etc. are Linux and not Aandroid. You have to call it from a SuperUser terminal process. Thus the commands:

Process chperm = Runtime.getruntime().exec("su");
and:
DataOutputStream os = new DataOutputStream(chperm.getOutputStream());  
requited.

The 1st obtains the reference to the SU process and the 2nd prepares output channel for the commands.

I'm going to implement another open-source project soon. The Systemyzer. 
I can drop you my current sources. Send me your email to my account
.
sms2000@runbox.com
